plugins/flashrom/fu-flashrom-device.c

```diff
@@ -23,12 +23,12 @@ struct _FuFlashromDevice {
 	FuUdevDevice parent_instance;
 	FuIfdRegion region;
 	struct flashrom_flashctx *flashctx;
-	struct flashrom_programmer *flashprog;
+	struct flashrom_layout *layout;
```

Comment:
Why do we need to store the flashrom_layout in the device? It doesn't seem to me we use it anywhere.

Reply:
Ohh i see, we use it during ->open() to set on the context; ignore me.

Reply:
Nope, we clear it during ->close() -- I think my original point holds.

Reply:
I just needed to store it somewhere so it doesn't leak. `flashrom_layout_set()` doesn't transfer ownership:
```c
/**
 * @brief Set the active layout for a flash context.
 *
 * Note: This just sets a pointer. The caller must not release the layout
 *       as long as he uses it through the given flash context.
 *
 * @param flashctx Flash context whose layout will be set.
 * @param layout   Layout to bet set.
 */
void flashrom_layout_set(struct flashrom_flashctx *const flashctx, const struct flashrom_layout *const layout)
{
	flashctx->layout = layout;
}
```

And there is no `flashrom_layout_get()`, so you can't extract it from the context to free later.

Reply:
> flashrom_layout_set() doesn't transfer ownership

Bleh. Okay, thanks for the explanation. Could you put that rationale in the source code please; one line is fine.